src/common/src/main/scala/osmesa/common/functions/osm/package.scala

```diff
@@ -499,93 +506,115 @@ package object osm {
     }
   }
 
-  private def connectSegments(segments: GenTraversable[Line])(implicit geometryFactory: GeometryFactory): GenTraversable[Line] =
-    connectSegments(segments.map(_.jtsGeom.getCoordinateSequence).map(s => new VirtualCoordinateSequence(Seq(s))))
+  private def connectSegments(segments: GenTraversable[jts.Geometry])(implicit geometryFactory: GeometryFactory): GenTraversable[LineString] =
```

Comment:
This no longer calls `connectSegments(segments: GenTraversable[VirtualCoordinateSequence], lines: Seq[CoordinateSequence] = Vector.empty[CoordinateSequence]): GenTraversable[CoordinateSequence]`, which explains some of the discrepancies (routes represented as `MultiLineString`s that could be `LineString`s).

Reply:
All of the geometry differences resolve after this diff:

```diff
diff --git i/src/common/src/main/scala/osmesa/common/functions/osm/package.scala w/src/common/src/main/scala/osmesa/common/functions/osm/package.scala
index 763680c..1253f67 100644
--- i/src/common/src/main/scala/osmesa/common/functions/osm/package.scala
+++ w/src/common/src/main/scala/osmesa/common/functions/osm/package.scala
@@ -507,12 +502,15 @@ package object osm {
   }
 
   private def connectSegments(segments: GenTraversable[jts.Geometry])(implicit geometryFactory: GeometryFactory): GenTraversable[LineString] =
-    segments
-      .map({ geom => if (classTag[jts.LineString].runtimeClass.isInstance(geom)) Some(geom.asInstanceOf[jts.LineString]) else None })
-      .flatten
+    connectSegments(
+      segments
+        .flatMap { geom =>
+        if (classTag[jts.LineString].runtimeClass.isInstance(geom))
+          Some(geom.asInstanceOf[jts.LineString])
+        else None
+      }
       .map(_.getCoordinateSequence)
-      .map(s => new VirtualCoordinateSequence(Seq(s)))
-      .map(geometryFactory.createLineString)
+      .map(s => new VirtualCoordinateSequence(Seq(s)))).map(geometryFactory.createLineString)
 
   private def getInteriorRings(p: jts.Polygon): Seq[jts.LinearRing] = for (i <- 0 until p.getNumInteriorRing) yield
     geometryFactory.createLinearRing(p.getInteriorRingN(i).getCoordinates)
```

Remaining inconsistencies are a result of differences in how `st_asText` is implemented (whatever) and the new version returning empty `MultiLineString`s (which is probably fine):

```
+{"id":1966314,"version":45,"minorVersion":0,"wkt":"MULTILINESTRING EMPTY"}
+{"id":1966314,"version":45,"minorVersion":0,"wkt":"MULTILINESTRING EMPTY"}
+{"id":1966314,"version":45,"minorVersion":0,"wkt":"MULTILINESTRING EMPTY"}

-{"id":2668952,"version":89,"minorVersion":0,"wkt":""}
+{"id":2668952,"version":89,"minorVersion":0,"wkt":null}

+{"id":3518908,"version":28,"minorVersion":0,"wkt":"MULTILINESTRING EMPTY"}

-{"id":4121287,"version":824,"minorVersion":0,"wkt":""}
+{"id":4121287,"version":824,"minorVersion":0,"wkt":null}

-{"id":58446,"version":148,"minorVersion":0,"wkt":""}
+{"id":58446,"version":148,"minorVersion":0,"wkt":null}

-{"id":58447,"version":396,"minorVersion":0,"wkt":""}
+{"id":58447,"version":396,"minorVersion":0,"wkt":null}

-{"id":62149,"version":725,"minorVersion":0,"wkt":""}
+{"id":62149,"version":725,"minorVersion":0,"wkt":null}

+{"id":6901530,"version":1,"minorVersion":0,"wkt":"MULTILINESTRING EMPTY"}
+{"id":6901654,"version":2,"minorVersion":0,"wkt":"MULTILINESTRING EMPTY"}
+{"id":7690203,"version":1,"minorVersion":0,"wkt":"MULTILINESTRING EMPTY"}
```